testing/unittests/dalitests.cpp

```diff
@@ -3265,7 +3265,150 @@ class DaliSysInfoLoggerTester : public CppUnit::TestFixture
+            fdesc->setDefaultDir("thordata/" SCOPE);
+
+            // Set file properties
+            fdesc->queryProperties().setPropInt("@recordSize", 17);
+            fdesc->queryProperties().setPropInt("@fileType", 1); // Flat file
+
+            // Create the distributed file
+            Owned<IDistributedFile> dfile = dir.createNew(fdesc);
+            ASSERT(dfile && "Failed to create distributed file");
+
+            // Attach the file to the logical name
+            dfile->attach(getSubFileLogicalName(subfileNum).c_str(), user);
+
+            ASSERT(dir.exists(getSubFileLogicalName(subfileNum).c_str(), user) && "Subfile was not created successfully");
+        }
+
+        // Create the superfile and add subfiles to the superfile
+        Owned<IDistributedSuperFile> superfile = dir.createSuperFile(getSuperFileLogicalName(1).c_str(), user, true, false);
+        ASSERT(superfile && "Failed to create superfile");
+        for (unsigned subfileNum = 1; subfileNum <= 3; subfileNum++)
+            superfile->addSubFile(getSubFileLogicalName(subfileNum).c_str());
+        superfile.clear();
+
+        // Create the supersuperfile and add superfile to it
+        Owned<IDistributedSuperFile> supersuperfile = dir.createSuperFile(getSuperFileLogicalName(2).c_str(), user, true, false);
+        ASSERT(supersuperfile && "Failed to create supersuperfile");
+        supersuperfile->addSubFile(getSuperFileLogicalName(1).c_str());
+        supersuperfile.clear();
+
+        // Some quick integrity checks
+        supersuperfile.setown(dir.lookupSuperFile(getSuperFileLogicalName(2).c_str(), user, AccessMode::readMeta));
+        ASSERT(supersuperfile && "Failed to find created supersuperfile");
+        unsigned numSubFiles = supersuperfile->numSubFiles();
+        ASSERT(numSubFiles == 1 && "Supersuperfile does not contain expected number of subfiles");
+
+        superfile.setown(dir.lookupSuperFile(getSuperFileLogicalName(1).c_str(), user, AccessMode::readMeta));
+        ASSERT(superfile && "Could not find created superfile");
+        numSubFiles = superfile->numSubFiles();
+        ASSERT(numSubFiles == 3 && "Superfile does not contain expected number of subfiles");
+        size32_t superRecordSize = (size32_t)superfile->queryAttributes().getPropInt("@recordSize", -1);
+        ASSERT(superRecordSize == 17 && "Superfile record size does not match subfiles");
+    }
+
+public:
+    ~FileReadPropertiesUpdaterTester()
+    {
+        daliClientEnd();
+    }
+    void testInit()
+    {
+        daliClientInit();
+    }
+
+    void testFileReadPropertiesUpdater()
+    {
+        cleanupTestFiles(false);
+        createTestFiles();
+
+        unsigned expectedSuperNumReads = 0;
+        unsigned expectedSuperReadCost = 0;
+        {
+            Owned<IFileReadPropertiesUpdater> fileReadPropertiesUpdater = createFileReadPropertiesUpdater(user);
+            for (unsigned subfileNum = 1; subfileNum <= 3; subfileNum++)
```

Comment:
would be good to avoid these constants ('3') and based on vector so easy to change/flexible.